packages/appkit/src/connectors/WalletConnectConnector.ts

```diff
@@ -43,7 +44,12 @@ export class WalletConnectConnector<Namespace extends ChainNamespace = ChainName
         caipNetworks,
         universalProviderConfigOverride
       )
-      await this.provider.connect({ optionalNamespaces: namespaces })
+      const res = await this.provider.connect({ optionalNamespaces: namespaces })
+
+      const chain = res?.namespaces?.[activeChain as ChainNamespace]?.chains?.[0]
```

Comment:
We should handle `activeChain`'s `undefined` state I think instead of type casting (`as ChainNamespace`)